pkg/security/module/module.go

```diff
@@ -119,6 +120,12 @@ func (m *Module) Start() error {
 		return err
 	}
 
+	if m.selfTester != nil {
+		if err := m.selfTester.CreateTargetFile(); err != nil {
```

Comment:
I might be wrong, but I don't think you need to do it before the reload anymore (I'm trying to keep the main functions of `module` as short as possible). I think you can put everything insite one single `selfTester.RunSelfTest()` function.

Reply:
The thing is we need to create the target file, to build the actual rules needed in the `Reload` function. But I'm completely open to find a better place to create those files, even maybe embed this into a `m.selfTester.Init` function that leaks less of the underlying implementation

Reply:
Oh right, because the directory name is randomly generated. Never mind then 👍🏻

Reply:
Actually I moved it from the `Start` as it's still an implementation detail: https://github.com/DataDog/datadog-agent/pull/8479/commits/fe57aa559d142f4b1e286b2c5d6d407780921f38

Reply:
👍🏻